src/main/java/duke/DukeList.java

```diff
@@ -0,0 +1,258 @@
+                break;
+            } catch (DukeNoDescriptionException e) {
+                throw new DukeInvalidDescriptionException(String.format("OOPS!!! The description of a `%s` cannot be empty.", taskType));
+            }
+
+        case DEADLINE:
+            try {
+                formattedDescriptionString = Parser.getItemSubstring(strArr);
+                newTask = new Deadline(formattedDescriptionString);
+                break;
+            } catch (DukeNoDescriptionException e) {
+                throw new DukeInvalidDescriptionException(String.format("OOPS!!! The description of a `%s` cannot be empty.", taskType));
+            } catch (DukeNoDateException e) {
+                throw new DukeInvalidDescriptionException(String.format("OOPS!!! The description of `%s` is invalid.", taskType));
+            }
+
+        case EVENT:
+            try {
+                formattedDescriptionString = Parser.getItemSubstring(strArr);
+                newTask = new Event(formattedDescriptionString);
+                break;
+            } catch (DukeNoDescriptionException e) {
+                throw new DukeInvalidDescriptionException(String.format("OOPS!!! The description of a `%s` cannot be empty.", taskType));
+            } catch (DukeNoDateException e) {
+                throw new DukeInvalidDescriptionException(String.format("OOPS!!! The description of `%s` is invalid.", taskType));
+            }
+
+        default:
+            String invalidCommand = strArr[0];
+            throw new DukeInvalidCommandException(String.format("OOPS!!! I'm sorry, but I don't know what `%s` means :-(", invalidCommand));
+        }
+
+        this.taskList.add(newTask);
+
+        return newTask;
+    }
+
+
+    /**
+     * Adds a new item to the list.
+     *
+     * @param descriptionString String to be added.
+     * @return Status string to be printed.
+     * @throws DukeException Duke exception.
+     */
+    public String add(String descriptionString) throws DukeException {
+        Task newTask = addHelper(descriptionString);
+
+        return "Got it. I've added this task:\n" +
+                String.format("\t%s\n", newTask.toString()) +
+                String.format("%s", this.getListStats());
+    }
+
+
+    /**
+     * Adds a new item to the list.
+     * This method is only accessed privately.
+     *
+     * @param descriptionString Description string of task to be added.
+     * @param isDone            If task is done already.
+     * @throws DukeInvalidDescriptionException invalid description text.
+     * @throws DukeInvalidCommandException     invalid command.
+     */
+    private void add(String descriptionString, boolean isDone) throws DukeInvalidDescriptionException, DukeInvalidCommandException {
+        Task newTask = addHelper(descriptionString);
+
+        if (isDone) {
+            newTask.markAsDone();
+        }
+    }
+
+
+    /**
+     * Marks an item as done.
```

Comment:
Can consider changing to "Marks an item as completed" for clarity and better english. But its really a matter of preference here (I prefer to use completed rather than done)